code/modules/mob/mob.dm

```diff
@@ -101,8 +101,10 @@
 // blind_message (optional) is what blind people will hear e.g. "You hear something!"
 // vision_distance (optional) define how many tiles away the message can be seen.
 // ignored_mob (optional) doesn't show any message to a given mob if TRUE.
+// agnosia_message gives a redacted version of the message to those who suffer from agnosia, to avoid identifying people's names
+// agnosia_self_message is the same, for the self_message
 
-/atom/proc/visible_message(message, self_message, blind_message, vision_distance, ignored_mob)
+/atom/proc/visible_message(message, self_message, blind_message, vision_distance, ignored_mob, agnosia_message, agnosia_self_message)
```

Comment:
This is quite weird. This is a very well used proc, and two new arguments for a trauma is undesirable. Unfortunately I don't think there is better way to achieve this otherwise without dynamically changing the names of the subjects dependent of recipient.

Reply:
I have to agree with these, these are horribly intrusive and these systems shouldn't need to know about something as specific as brain traumas to work properly.